modules/virt-creating-custom-monitoring-label-for-vms.adoc

```diff
@@ -0,0 +1,64 @@
+// Module included in the following assemblies:
+//
+// * virt/logging_events-monitoring/virt-exposing-custom-metrics-for-vms.adoc
+
+[id="virt-creating-custom-monitoring-label-for-vms_{context}"]
+= Creating a custom monitoring label for virtual machines
+
+To enable queries to multiple virtual machines from a single service, create a custom label and assign it to the virtual machine's YAML file.
+
+.Prerequisites
+
+* Install the {product-title} CLI `oc`.
+* Log in as a user with `cluster-admin` privileges.
+* Access to the web console for stopping and restarting a virtual machine.
+
+.Procedure
+* Edit the `template` spec of your virtual machine configuration file. The following example shows the label name `metrics` being added to the label `monitor`.
++
+[source,yaml]
+----
+spec:
+  template:
+    metadata:
+      labels:
+        monitor: metrics
+----
+
+* Stop and restart the virtual machine to create a new pod with the label name given to the `monitor` label.
+
+.Verification
+
+Metrics are exposed through an HTTP service endpoint under the `/metrics` canonical name. To verify that the `node-exporter` service is working:
+
+. Obtain the `Cluster-IP` for the `node-exporter` service.
++
+[source,terminal]
+----
+$ oc get service -n <node_exporter_service_name>
+----
++
+.Example output
+[source,terminal]
+----
+NAME                  TYPE      CLUSTER-IP     EXTERNAL-IP PORT(S)  AGE
+node-exporter-service ClusterIP 172.30.226.162 <none>      9100/TCP 4d22h
+
+----
+
+. To list all available metrics for the `node-exporter` service, use the `curl` query against `http://<endpoint>/metrics`.
++
+[source,terminal]
+----
+$ curl http://<172.30.226.162:9100>/metrics
```

Comment:
In order to run this command we have to expose the service, direct using Cluster-IP is not allowed.
@jobbler can you please suggest here.?

Reply:
im curious why this is not allowed? technical limitation or other?

Reply:
@sbulage , @jobbler - Please see my Google chat message - would appreciate if you can respond in the Google chat about having a meeting to resolve this discussion.

Reply:
Here are the steps. Not much to it. You may need to change the following a little to match your document - like hostnames, service names and VM names.

Expose the service:
```
$ oc expose service node-exporter-service
[route.route.openshift.io/node-exporter-service](http://route.route.openshift.io/node-exporter-service) exposed
```

Get the FQDN of the route
```
$ oc get route -o=custom-columns=NAME:.[metadata.name](http://metadata.name/),DNS:.spec.host
NAME                    DNS
node-exporter-service   [node-exporter-service-dynamation.apps.cluster.example.org](http://node-exporter-service-dynamation.apps.cluster.example.org/)
```

Pull the metrics from outside the cluster. Note you dont need the port number.
```
$ curl -s http://node-exporter-service-dynamation.apps.cluster.example.org/metrics
```
Please note that this pulls the metrics from one system at a time if there are multiple on the service. So the user will need to run the command multiple times to get all systems.

An obvious way around this would be for the user to create a service for each VM. It all depends on their use case.

I dont think you would need to go into much detail on this other than note to the user they will need to make multiple requests if the service pulls from multiple VMs.

Reply:
In the Verification section, we had 2 steps. I replaced those 2 steps with the 3 steps noted here. Please review and advise.

Reply:
@sbulage - Please review and approve the updated content by adding LGTM to this GitHub issue.